### PR TITLE
BUG: fix rebin_piecewise_constant

### PR DESCRIPTION
**Pull request by andyfaff**
This passes the tests, as well as the case presented in #9. It's worth getting a second set of eyes on the changes I made, to check the logic.

### CODE REVIEW

File: rebin.py [hunk 1]
```diff
@@ -203,113 +203,44 @@ def rebin_spline(x1, y1, x2, interp_kind):
...
+    # greater than or equal to one original bin.
+    # This is the contribution from the 'intact' bins (not including the
+    # fractional start and end parts.
+    whole_bins = np.floor(i_place[1:]) - np.ceil(i_place[:-1]) >= 1.
+    start = cum_sum[np.ceil(i_place[:-1]).astype(int)]
+    finish = cum_sum[np.floor(i_place[1:]).astype(int)]
 
-        # deal with whole parts of bin that are spanned
-        cum_sum = np.cumsum(y1)
-        running_sum = (cum_sum[i_place[start_pos + 1:end_pos + 1] - 2]
```
Comment (buroojg) [inline rebin.py]: There is an error in this line: 

the result of this [i_place[start_pos + 1:end_pos + 1] - 2] should be a boolean array. or? But this generates an integer array.

Comment (andyfaff) [reply rebin.py]: The `-` signs are an indication that the line is being removed.

Comment (andyfaff) [reply rebin.py]: The PR is replacing the entire function.

Comment (jhykes) [reply rebin.py]: @andyfaff Care to comment on @buroojg's question?

**Comment by jhykes** [comment]
Thanks @andyfaff for fixing this!

**Comment by bghani** [comment]
Thanks a lot @andyfaff for fixing the code!

**Pull request merged**